### ISSUE TITLE
Accuracy Drop in ROUGE Scores After Merging Mistral 7B Model for CNN/DailyMail Finetuning

### ISSUE THREAD

**Issue by xihajun** [huggingface/autotrain-advanced#443]
Hello, I am experiencing a significant drop in ROUGE scores when merging a finetuned Mistral 7B model for the CNN/DailyMail summarization task. I would appreciate any insights or assistance in understanding and resolving this issue.

## Description:
Hello, I am experiencing a significant drop in ROUGE scores when merging a finetuned Mistral 7B model for the CNN/DailyMail summarization task. I would appreciate any insights or assistance in understanding and resolving this issue.

## Steps to Reproduce:
1. Finetuned `mistralai/Mistral-7B-v0.1` on CNN/DailyMail dataset (the finetuned adapter could be found `xihajun/new_cnn_test`) without using the `--auto-merge` option.
2. Evaluated it, receiving the following ROUGE scores:
   - ROUGE-1: 0.3518
   - ROUGE-2: 0.1557
   - ROUGE-L: 0.2533
   - ROUGE-Lsum: 0.3170

The way I load the model is using `xihajun/new_cnn_test` as `model_name` with `autotrain-advanced` installed
```
    def load_model_and_tokenizer(self):
        self.model = AutoModelForCausalLM.from_pretrained(self.model_name).to(self.device)
        self.tokenizer = AutoTokenizer.from_pretrained(
            self.model_name,
            model_max_length=2048,
            padding_side="left",
            use_fast=False
        )
        self.tokenizer.padding_side = "left"
        self.tokenizer.pad_token = self.tokenizer.eos_token
        self.tokenizer.model_max_length = 2048
```

3. Merged the model using two methods:
   - Hugging Face Space: [llm-merge-adapter](https://huggingface.co/spaces/autotrain-projects/llm-merge-adapter)
   - Manual merging as per the provided [script](https://github.com/huggingface/autotrain-advanced/issues/299#issuecomment-1871315142).
     - maybe I should not set `torch_dtype=torch.float16,`?
     - as [mistral](https://huggingface.co/mistralai/Mistral-7B-v0.1/blob/26bca36bde8333b5d7f72e9ed20ccda6a618af24/config.json#L20) 7b: `"torch_dtype": "bfloat16",`

<details>

```
from transformers import AutoModelForCausalLM, AutoTokenizer, AutoConfig
import torch
from peft import PeftModel
adapter_path="xxx"
base_model_path="mistralai/Mistral-7B-v0.1"
target_model_path="model_output/"
config = AutoConfig.from_pretrained(base_model_path)


model = AutoModelForCausalLM.from_pretrained(
    base_model_path,
    torch_dtype=torch.float16,
    low_cpu_mem_usage=True,
    trust_remote_code=True,
)

model = PeftModel.from_pretrained(model, adapter_path)

tokenizer = AutoTokenizer.from_pretrained(
    base_model_path,
    trust_remote_code=True,
)
model = model.merge_and_unload()

logger.info("Saving target model...")
model.save_pretrained(target_model_path)
tokenizer.save_pretrained(target_model_path)
config.save_pretrained(target_model_path)
```
</details>

4. After merging, the ROUGE scores dropped significantly:
   - ROUGE-1: 0.2630
   - ROUGE-2: 0.1155
   - ROUGE-L: 0.1726
   - ROUGE-Lsum: 0.2123

## Expected Behavior:
`ROUGE-1 ~ 0.3518`

## Related Issues
#299

**Comment by abhishekkrthakur** [comment]
Could you please add training parameters too?

**Comment by xihajun** [comment]
> Could you please add training parameters too?

Thanks for your reply @abhishekkrthakur May I ask where can I add those parameters?

**Comment by abhishekkrthakur** [comment]
anywhere. in the original post or as a comment

**Comment by xihajun** [comment]
@abhishekkrthakur Ah, thanks for pointing out, I think I forgot I have already finetuned the model, and the base model should be the finetuned one :)

Let me check if that would solve this issue

**Comment by xihajun** [comment]
now it worked better now, thanks I will close the issue

**Comment by abhishekkrthakur** [comment]
what was the issue? :)

**Comment by xihajun** [comment]
> what was the issue? :)

I merged to the wrong base model :)

**Comment by abhishekkrthakur** [comment]
glad you figured it out :)

**Comment by xihajun** [comment]
Dear @abhishekkrthakur, may I ask if that is normal if both `torch.float16` and `bfloat16` returns the same accuracy?
